kernel/src/platform/watchdog.rs

```diff
@@ -0,0 +1,27 @@
+//! Interface for configuring a watchdog
+
+/// A trait for implementing a watchdog in the kernel.
+/// This trait is called from the `kernel_loop()` code to setup
+/// and maintain the watchdog timer.
+/// It is up to the specific `Chip` how it will handle watchdog interrupts.
+pub trait WatchDog {
+    /// This function must enable the watchdog timer and configure it to
+    /// trigger regulary. The period of the timer is left to the implementation
+    /// to decide. The implementation must ensure that it doesn't trigger too
+    /// early (when we haven't hung for example) or too late as to not catch
+    /// faults.
+    fn setup(&self) {}
+
+    /// This function must tickle the watchdog to reset the timer.
+    /// If the watchdog was previously suspended then this should also
+    /// resume the timer.
+    fn tickle(&self) {}
+
+    /// Suspends the watchdog timer. After calling this the timer should not
+    /// fire until after `tickle()` has been called. This function is called
+    /// before sleeping.
+    fn suspend(&self) {}
+}
+
+/// Implement default WatchDog trait for unit.
+impl WatchDog for () {}
```

Comment:
There is already a watchdog trait in kernel/src/hil/watchdog.rs that is implemented by the sam4l (though not currently used in the kernel). It probably makes sense to modify that one rather than have two different watchdog traits.
Or just remove that one in favor of this one, and change the sam4l to implement the new one.

Reply:
Before I go ahead and convert it over to a new implementation I would like to have the implementation agreed upon. Once that's done I'll update the SAM4L